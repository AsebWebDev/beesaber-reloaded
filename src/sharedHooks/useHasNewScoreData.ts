import { skipToken } from '@reduxjs/toolkit/dist/query';
import { useEffect, useState } from 'react';

import { useGetFullPlayerQuery } from '@/api/services/apiPlayer/apiPlayer';
import { useGetUserDataQuery } from '@/api/services/apiUser/apiUser';

const useHasNewScoreData = (userId: string): boolean => {
  const [scoreSaberId, setScoreSaberId] = useState<string | undefined>(
    undefined
  );
  const [hasNewScores, setHasNewScores] = useState<boolean>(false);

  const { data } = useGetUserDataQuery(userId);

  const _scoreSaberId = data?.playerInfo?.playerId;
  const totalPlayCount = data?.totalPlayCount;

  useEffect(() => {
    if (_scoreSaberId !== undefined) setScoreSaberId(_scoreSaberId);
  }, [_scoreSaberId]);

  const { data: scoreSaberData } = useGetFullPlayerQuery(
    scoreSaberId ?? skipToken
  );

  useEffect(() => {
    const totalPlayCountSS = scoreSaberData?.scoreStats.totalPlayCount;

    if (
      totalPlayCount !== undefined &&
      totalPlayCountSS !== undefined &&
      totalPlayCount < totalPlayCountSS
    )
      setHasNewScores(true);
  }, [scoreSaberData]);

  return hasNewScores;
};

export default useHasNewScoreData;
