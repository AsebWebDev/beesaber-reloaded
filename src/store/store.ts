import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/dist/query/react';

import { apiInitialState } from '@/api/api';
import { apiAuth } from '@/api/services/apiAuth/apiAuth';
import { apiPlayer } from '@/api/services/apiPlayer/apiPlayer';
import { apiUser } from '@/api/services/apiUser/apiUser';

import { initialState as appStatusInitialState } from './reducer/appStatusReducer';
import { initialState as notificationsInitialState } from './reducer/notificationsReducer';
import rootReducer from './reducer/rootReducer';
import { initialState as userDataInitialState } from './reducer/userDataReducer';

const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(
      apiAuth.middleware,
      apiPlayer.middleware,
      apiUser.middleware
    ),
  devTools: true,
});

setupListeners(store.dispatch);

const initialState = {
  appStatus: { ...appStatusInitialState },
  notifications: { ...notificationsInitialState },
  userData: { ...userDataInitialState },
  api: { ...apiInitialState },
};

export { initialState };

export default store;

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
