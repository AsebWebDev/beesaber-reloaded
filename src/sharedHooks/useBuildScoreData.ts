/* eslint-disable no-console */
import { useEffect, useState } from 'react';
import { toast } from 'react-toastify';

import getAllScores from '@/api/services/helper/getAllScores';
import { useAppDispatch } from '@/store/hooks';
import { userIsFetchingData } from '@/store/reducer/appStatusReducer';

import type { ScoreData } from '@/../sharedTypes';

/**
 * Fetch and parse all score data of a user
 * @param {String} id - ScoreSaber id
 * @param {number} threshold - optional threshold for number of score pages (1 page = 8 scores)
 * @returns {ScoreData} score data object with recent scores, top scores and a list of
 * score hashes
 */
const useBuildScoreData = (
  id?: string,
  threshold?: number
): ScoreData | undefined => {
  if (id === undefined) return undefined;
  const dispatch = useAppDispatch();
  const [isFetching, setIsFetching] = useState(false);
  const [scoreData, setScoreData] = useState<ScoreData>({
    scoredSongsHashes: [],
    scoresRecent: [],
    scoresTop: [],
  });

  useEffect(() => {
    setIsFetching(true);
    const fetchData = async () => {
      try {
        const resp = await toast.promise<ScoreData>(
          getAllScores(id, threshold),
          {
            pending: 'Updating scores ...',
            success: 'We updated all Scores ðŸ‘Œ',
            error: 'There has been an issue your scores ðŸ¤¯',
          }
        );

        setScoreData(resp);
        setIsFetching(false);
      } catch (error: unknown) {
        console.log(
          'ðŸš€ ~ file: useBuildScoreData.ts ~ line 34 ~ fetchData ~ error',
          error
        );
        setIsFetching(false);
      }
    };

    void fetchData();
  }, [id]);

  dispatch(userIsFetchingData({ status: isFetching }));

  return scoreData;
};

export default useBuildScoreData;
