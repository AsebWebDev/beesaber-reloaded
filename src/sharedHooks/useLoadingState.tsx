import { skipToken } from '@reduxjs/toolkit/dist/query/react';

import { useGetUserDataQuery } from '@/api/services/apiUser/apiUser';
import { useAppSelector } from '@/store/hooks';
import { selectUserId } from '@/store/reducer/userDataReducer';

const useLoadingState = (): boolean => {
  const userId = useAppSelector(selectUserId);
  const { isFetching: isFetchingUserData } = useGetUserDataQuery(
    userId ?? skipToken
  );

  return isFetchingUserData;
};

export default useLoadingState;
