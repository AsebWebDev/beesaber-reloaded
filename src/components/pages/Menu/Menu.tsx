import { Link } from 'react-router-dom';
import styled from 'styled-components';

import BrandLogo from '@/components/common/BrandLogo/BrandLogo';
import GoogleOAuth from '@/components/common/GoogleOAuth/GoogleOAuth';
import Title from '@/components/common/Title/Title';
import Divider from '@/components/tools/Divider/Divider';
import parseId from '@/helper/parseEmptyStringToUndefined';
import useIsMobile from '@/sharedHooks/useIsMobile';
import useLoadingState from '@/sharedHooks/useLoadingState';
import { useAppSelector } from '@/store/hooks';
import { selectMyScoreSaberId } from '@/store/reducer/userDataReducer';
import { mediaQuery } from '@/tokens/definitions/layout';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  font-family: 'NeonTubes2';
  padding: 1rem 1rem 0 1rem;
  min-height: 100vh;

  ${mediaQuery.sm} {
    padding: 1rem;
    height: 100%;
  }
`;

const MenuContainer = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
  padding-bottom: 9rem;

  ${mediaQuery.sm} {
    padding-bottom: 3rem;
  }
`;

const MenuHeader = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  width: 100%;
  margin-bottom: 3%;
`;

const MenuLink = styled(Link)`
  display: flex;
`;

const MenuItems = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
`;

const Menu = (): JSX.Element => {
  const { isMobile } = useIsMobile();
  const isLoading = useLoadingState();
  const myScoreSaberId = useAppSelector(selectMyScoreSaberId);

  return (
    <Container data-testid={'menu'}>
      <MenuContainer>
        <MenuHeader>
          <Title>Bee Saber</Title>
          <BrandLogo />
        </MenuHeader>
        <MenuItems>
          <MenuLink to="/">
            <Title>Dashboard</Title>
          </MenuLink>
          <MenuLink to="/myprofile">
            <Title>My Profile</Title>
          </MenuLink>
          {parseId(myScoreSaberId) !== undefined && !isLoading && (
            <MenuLink to="/myhive">
              <Title>My Hive</Title>
            </MenuLink>
          )}
        </MenuItems>
        <GoogleOAuth />
      </MenuContainer>
      {isMobile && <Divider />}
    </Container>
  );
};

export default Menu;
