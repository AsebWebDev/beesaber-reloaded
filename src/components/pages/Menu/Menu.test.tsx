import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import * as useIsMobile from '@/sharedHooks/useIsMobile';
import { initialState as store } from '@/store/store';

import Menu from './Menu';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('components/pages/Menu', () => {
  let spy: jest.Mock | jest.SpyInstance<{ isMobile: boolean }, []>;

  beforeEach(() => {
    spy = jest.fn();
    spy = jest.spyOn(useIsMobile, 'default');
  });

  afterEach(() => {
    spy.mockRestore();
    spy.mockReset();
  });

  it.each([true, false])(
    'should match snapshot when isMobile is %sw',
    (isMobile) => {
      spy.mockReturnValue({ isMobile });

      const { container } = render(
        <Router>
          <Provider store={mockStore(store)}>
            <Menu />
          </Provider>
        </Router>
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );
});
