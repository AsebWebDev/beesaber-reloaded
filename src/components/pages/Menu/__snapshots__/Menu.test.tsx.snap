// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Menu should match snapshot when isMobile is falsew 1`] = `
.c7 {
  font-family: 'NeonTubes2';
}

.c8 {
  color: #fffcdd;
  text-shadow: #ffeb00 -2px -1px 5px,#ffeb00 2px 1px 5px, #ffeb00 0px 0px 10px,#ffeb00 0px 0px 15px, #ffeb00 0px 0px 20px,#ffeb00 0px 0px 30px;
}

.c6 {
  position: relative;
  height: 150px;
  width: 150px;
  font-size: 80px;
  margin-bottom: 1rem;
}

.c6 div {
  position: absolute;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c6 div:first-child {
  -webkit-transform: rotate(90deg) translate(-36px,-68px);
  -ms-transform: rotate(90deg) translate(-36px,-68px);
  transform: rotate(90deg) translate(-36px,-68px);
  z-index: 5;
}

.c6 div:nth-child(2) {
  -webkit-transform: rotate(150deg) translate(-103px,-49px);
  -ms-transform: rotate(150deg) translate(-103px,-49px);
  transform: rotate(150deg) translate(-103px,-49px);
  z-index: 4;
}

.c6 div:nth-child(3) {
  -webkit-transform: rotate(30deg) translate(120px,-19px);
  -ms-transform: rotate(30deg) translate(120px,-19px);
  transform: rotate(30deg) translate(120px,-19px);
}

.c6 div:nth-child(4) {
  -webkit-transform: rotate(90deg) translate(70px,-68px);
  -ms-transform: rotate(90deg) translate(70px,-68px);
  transform: rotate(90deg) translate(70px,-68px);
}

.c6 div:nth-child(5) {
  -webkit-transform: rotate(150deg) translate(2px,-49px);
  -ms-transform: rotate(150deg) translate(2px,-49px);
  transform: rotate(150deg) translate(2px,-49px);
}

.c6 div:nth-child(6) {
  -webkit-transform: rotate(30deg) translate(15px,-20px);
  -ms-transform: rotate(30deg) translate(15px,-20px);
  transform: rotate(30deg) translate(15px,-20px);
  z-index: 6;
}

.c6 .saber {
  position: absolute;
  height: 100%;
  width: 3px;
  top: 0%;
  border-radius: 3px;
}

.c6 .saber.neon-red {
  left: 49%;
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg);
  background-color: #fff0f0;
  box-shadow: 0px 0px 8px 2px #ff8d9e, 0px 0px 12px 6px #ff0025;
}

.c6 .saber.neon-blue {
  right: 49%;
  -webkit-transform: rotate(-30deg);
  -ms-transform: rotate(-30deg);
  transform: rotate(-30deg);
  background-color: #edf5ff;
  box-shadow: 0px 0px 8px 2px #ff8d9e, 0px 0px 12px 6px #ff0025;
}

.c4 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c5 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #0000FF -2px -1px 5px,#0000FF 2px 1px 5px, #0000FF 0px 0px 10px,#0000FF 0px 0px 15px,#0000FF 0px 0px 20px, #0000FF 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  font-family: 'NeonTubes2';
  padding: 1rem 1rem 0 1rem;
  min-height: 100vh;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 9rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin-bottom: 3%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

@media (min-width:576px) {
  .c0 {
    padding: 1rem;
    height: 100%;
  }
}

@media (min-width:576px) {
  .c1 {
    padding-bottom: 3rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1
          class="c4"
        >
          Bee
        </h1>
        <h1
          class="c5"
        >
          Saber
        </h1>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
      </div>
    </div>
    <div
      class="c9"
    >
      <a
        class="c10"
        href="/"
      >
        <div
          class="c3"
        >
          <h1
            class="c4"
          >
            Dashboard
          </h1>
        </div>
      </a>
      <a
        class="c10"
        href="/myprofile"
      >
        <div
          class="c3"
        >
          <h1
            class="c4"
          >
            My
          </h1>
          <h1
            class="c5"
          >
            Profile
          </h1>
        </div>
      </a>
      <a
        class="c10"
        href="/myhive"
      >
        <div
          class="c3"
        >
          <h1
            class="c4"
          >
            My
          </h1>
          <h1
            class="c5"
          >
            Hive
          </h1>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`components/pages/Menu should match snapshot when isMobile is truew 1`] = `
.c7 {
  font-family: 'NeonTubes2';
}

.c8 {
  color: #fffcdd;
  text-shadow: #ffeb00 -2px -1px 5px,#ffeb00 2px 1px 5px, #ffeb00 0px 0px 10px,#ffeb00 0px 0px 15px, #ffeb00 0px 0px 20px,#ffeb00 0px 0px 30px;
}

.c6 {
  position: relative;
  height: 150px;
  width: 150px;
  font-size: 80px;
  margin-bottom: 1rem;
}

.c6 div {
  position: absolute;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c6 div:first-child {
  -webkit-transform: rotate(90deg) translate(-36px,-68px);
  -ms-transform: rotate(90deg) translate(-36px,-68px);
  transform: rotate(90deg) translate(-36px,-68px);
  z-index: 5;
}

.c6 div:nth-child(2) {
  -webkit-transform: rotate(150deg) translate(-103px,-49px);
  -ms-transform: rotate(150deg) translate(-103px,-49px);
  transform: rotate(150deg) translate(-103px,-49px);
  z-index: 4;
}

.c6 div:nth-child(3) {
  -webkit-transform: rotate(30deg) translate(120px,-19px);
  -ms-transform: rotate(30deg) translate(120px,-19px);
  transform: rotate(30deg) translate(120px,-19px);
}

.c6 div:nth-child(4) {
  -webkit-transform: rotate(90deg) translate(70px,-68px);
  -ms-transform: rotate(90deg) translate(70px,-68px);
  transform: rotate(90deg) translate(70px,-68px);
}

.c6 div:nth-child(5) {
  -webkit-transform: rotate(150deg) translate(2px,-49px);
  -ms-transform: rotate(150deg) translate(2px,-49px);
  transform: rotate(150deg) translate(2px,-49px);
}

.c6 div:nth-child(6) {
  -webkit-transform: rotate(30deg) translate(15px,-20px);
  -ms-transform: rotate(30deg) translate(15px,-20px);
  transform: rotate(30deg) translate(15px,-20px);
  z-index: 6;
}

.c6 .saber {
  position: absolute;
  height: 100%;
  width: 3px;
  top: 0%;
  border-radius: 3px;
}

.c6 .saber.neon-red {
  left: 49%;
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg);
  background-color: #fff0f0;
  box-shadow: 0px 0px 8px 2px #ff8d9e, 0px 0px 12px 6px #ff0025;
}

.c6 .saber.neon-blue {
  right: 49%;
  -webkit-transform: rotate(-30deg);
  -ms-transform: rotate(-30deg);
  transform: rotate(-30deg);
  background-color: #edf5ff;
  box-shadow: 0px 0px 8px 2px #ff8d9e, 0px 0px 12px 6px #ff0025;
}

.c4 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c5 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #0000FF -2px -1px 5px,#0000FF 2px 1px 5px, #0000FF 0px 0px 10px,#0000FF 0px 0px 15px,#0000FF 0px 0px 20px, #0000FF 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c11 {
  margin-top: 1rem;
  margin-bottom: 1rem;
  border: 0;
  border-top: 1px solid rgba(255,255,255,0.7);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  font-family: 'NeonTubes2';
  padding: 1rem 1rem 0 1rem;
  min-height: 100vh;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 9rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin-bottom: 3%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

@media (min-width:576px) {
  .c0 {
    padding: 1rem;
    height: 100%;
  }
}

@media (min-width:576px) {
  .c1 {
    padding-bottom: 3rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1
          class="c4"
        >
          Bee
        </h1>
        <h1
          class="c5"
        >
          Saber
        </h1>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            I
          </span>
        </div>
      </div>
    </div>
    <div
      class="c9"
    >
      <a
        class="c10"
        href="/"
      >
        <div
          class="c3"
        >
          <h1
            class="c4"
          >
            Dashboard
          </h1>
        </div>
      </a>
      <a
        class="c10"
        href="/myprofile"
      >
        <div
          class="c3"
        >
          <h1
            class="c4"
          >
            My
          </h1>
          <h1
            class="c5"
          >
            Profile
          </h1>
        </div>
      </a>
      <a
        class="c10"
        href="/myhive"
      >
        <div
          class="c3"
        >
          <h1
            class="c4"
          >
            My
          </h1>
          <h1
            class="c5"
          >
            Hive
          </h1>
        </div>
      </a>
    </div>
  </div>
  <hr
    class="c11"
  />
</div>
`;
